adapters/p2p2core/transaction.go

```
@@ -151,9 +182,12 @@ func AdaptL1Handler(tx *transaction.L1HandlerV0, txnHash *common.Hash) *core.L1H
 }
 
 //nolint:funlen
-func AdaptTransaction(t *synctransaction.TransactionInBlock, network *utils.Network) core.Transaction {
+func AdaptTransaction(
+	t *synctransaction.TransactionInBlock,
+	network *utils.Network,
+) (core.Transaction, error) {
 	if t == nil {
-		return nil
+		return nil, nil
 	}
```

Let's remove this check. I don't think it is a good idea to silently do nothing when we receive a nil pointer in general. This affects the code in `sync.go` so it will need updating to handle this change in the condition.

---

If we remove this check, won't the program crash at the following statement?
```
switch t.Txn.(type)
```

---

Removed the check.